### PR TITLE
feat: Rename the `wasmer-engine-native` to `wasmer-engine-shared-library`

### PR DESCRIPTION
**Pull request by Hywan**
# Description

This engine had a bad name, since “native” means everything and
nothing at the same time. This engine generates a shared object
library, so we decided to name it `wasmer-engine-shared-library`.

To do once this PR is merged:

- [ ] Update the documentation (docs.wasmer.io)
- [ ] Update the Github issue labels
- [ ] Update the language integrations
- [ ] Do something on crates.io to yank `wasmer-engine-native` probably?

# Review

- [x] Add a short description of the the change to the CHANGELOG.md file

### CODE REVIEW

**Comment by Hywan** [comment]
bors try

File: lib/cli/src/store.rs [hunk 1]
```diff
@@ -19,16 +19,16 @@ pub struct StoreOptions {
...
-    #[structopt(long, conflicts_with_all = &["native", "object_file"])]
+    /// Use the JIT Engine.
+    #[structopt(long, conflicts_with_all = &["shared_library", "object_file"])]
     jit: bool,
 
-    /// Use Native Engine.
+    /// Use the Shared Library Engine.
     #[structopt(long, conflicts_with_all = &["jit", "object_file"])]
-    native: bool,
+    shared_library: bool,
```
Comment (MarkMcCaskey) [inline lib/cli/src/store.rs:28]: Perhaps add an alias of `native`? Unsure what we want to do in regards to CLI flag stability

**Review by MarkMcCaskey — APPROVED** [review]
We'll need to put a task somewhere to publish under the old name once more (in a non-breaking way) marking it as deprecated and linking to the new one in the readme.

I'm not 100% on the new name but I do think `native` is a confusing name and I've probably gotten used to it, so I'd say this name change is probably an upgrade 👍

File: lib/cli/src/store.rs (hunk 1 above)
Comment (Hywan) [reply lib/cli/src/store.rs:28]: The 1.0 is not released yet, not sure we want to have compatibility between alpha, beta, and stable version 🤔.

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
bors try

**Comment by Hywan** [comment]
Closed in favor of #2356.

**Pull request closed without merging**